include/zephyr/drivers/led/lp50xx.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2020 Seagate Technology LLC
+ * Copyright (c) 2022 Grinn
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_LED_LP50XX_H_
+#define ZEPHYR_INCLUDE_DRIVERS_LED_LP50XX_H_
+
+#define LP50XX_COLORS_PER_LED	3
+
+#define LP5009_MAX_LEDS		3
+#define LP5012_MAX_LEDS		4
+#define LP5018_MAX_LEDS		6
+#define LP5024_MAX_LEDS		8
+#define LP5030_MAX_LEDS		10
+#define LP5036_MAX_LEDS		12
+
+/*
+ * For the rest of this file, the following macros are compatible with chips:
+ *   - LP5012_XXX => LP5009 / LP5012
+ *   - LP5024_XXX => LP5018 / LP5024
+ *   - LP5036_XXX => LP5030 / LP5036
+ */
+
+/* LED brightness channels. */
+#define LP5012_LED_BRIGHT_CHAN_BASE	0x07
```

Comment:
In the original `lp503x` driver these definitions are not actual register addresses but indexes starting at 0 with the first register exposed: "bank brightness".

```
/* Bank channels. */
#define LP503X_BANK_CHAN_BASE           0
#define LP503X_BANK_BRIGHT_CHAN         LP503X_BANK_CHAN_BASE
```

Again, we don't want this patch to break user apps. Let's save the API changes and discussions for another PR.

Reply:
OK, I just need to also fix the sample to return to the state before